File: lib/annotate.js
const fs = require('fs');
const path = require('path');
const os = require('os');

const AIGIS_DIR = path.join(os.homedir(), '.aigis');
const ANNOTATIONS_FILE = path.join(AIGIS_DIR, 'annotations.json');

function ensureDir() {
  if (!fs.existsSync(AIGIS_DIR)) {
    fs.mkdirSync(AIGIS_DIR, { recursive: true });
  }
}

function loadAnnotations() {
  if (!fs.existsSync(ANNOTATIONS_FILE)) return {};
  try {
    return JSON.parse(fs.readFileSync(ANNOTATIONS_FILE, 'utf8'));
  } catch {
    return {};
  }
}

function saveAnnotations(data) {
  ensureDir();
  fs.writeFileSync(ANNOTATIONS_FILE, JSON.stringify(data, null, 2));
}

function annotate(fileId, note) {
  const data = loadAnnotations();
  if (!data[fileId]) data[fileId] = [];
  data[fileId].push({
    note,
    date: new Date().toISOString().split('T')[0],
  });
  saveAnnotations(data);
}

function getAnnotations(fileId) {
  const data = loadAnnotations();
  return data[fileId] || [];
}

function listAnnotations() {
  return loadAnnotations();
}

function clearAnnotation(fileId) {
  const data = loadAnnotations();
  delete data[fileId];
  saveAnnotations(data);
}

module.exports = { annotate, getAnnotations, listAnnotations, clearAnnotation };

File: cli-python/aigis_cli/annotate.py
from __future__ import annotations

import json
from datetime import date
from pathlib import Path

AIGIS_DIR = Path.home() / ".aigis"
ANNOTATIONS_FILE = AIGIS_DIR / "annotations.json"


def _load_annotations() -> dict:
    if not ANNOTATIONS_FILE.exists():
        return {}
    try:
        return json.loads(ANNOTATIONS_FILE.read_text(encoding="utf-8"))
    except (json.JSONDecodeError, OSError):
        return {}


def _save_annotations(data: dict) -> None:
    AIGIS_DIR.mkdir(parents=True, exist_ok=True)
    ANNOTATIONS_FILE.write_text(json.dumps(data, indent=2), encoding="utf-8")


def annotate(file_id: str, note: str) -> None:
    data = _load_annotations()
    data.setdefault(file_id, []).append({
        "note": note,
        "date": date.today().isoformat(),
    })
    _save_annotations(data)


def get_annotations(file_id: str) -> list[dict]:
    return _load_annotations().get(file_id, [])


def list_annotations() -> dict:
    return _load_annotations()


def clear_annotation(file_id: str) -> None:
    data = _load_annotations()
    data.pop(file_id, None)
    _save_annotations(data)
